src/services/tx/txMonitor.ts

```diff
@@ -106,38 +111,45 @@ export const waitForRelayedTx = (taskId: string, txId: string): void => {
         timeoutId = setTimeout(checkTxStatus, POLLING_INTERVAL)
         return
       case TaskState.ExecSuccess:
-        txDispatch(TxEvent.PROCESSED, {
-          txId,
-        })
+        txId &&
+          txDispatch(TxEvent.PROCESSED, {
+            txId,
+          })
+        setStatus && setStatus(SafeCreationStatus.SUCCESS)
         clearTimeout(failAfterTimeoutId)
         return
       case TaskState.ExecReverted:
-        txDispatch(TxEvent.REVERTED, {
-          txId,
-          error: new Error(`Relayed transaction reverted by EVM.`),
-        })
+        txId &&
+          txDispatch(TxEvent.REVERTED, {
+            txId,
+            error: new Error(`Relayed transaction reverted by EVM.`),
+          })
         clearTimeout(failAfterTimeoutId)
         return
       case TaskState.Blacklisted:
-        txDispatch(TxEvent.FAILED, {
-          txId,
-          error: new Error(`Relayed transaction was blacklisted by relay provider.`),
-        })
+        txId &&
+          txDispatch(TxEvent.FAILED, {
+            txId,
+            error: new Error(`Relayed transaction was blacklisted by relay provider.`),
+          })
         clearTimeout(failAfterTimeoutId)
         return
       case TaskState.Cancelled:
-        txDispatch(TxEvent.FAILED, {
-          txId,
-          error: new Error(`Relayed transaction was cancelled by relay provider.`),
-        })
+        txId &&
```

Comment:
Need we even call `waitForRelayedTx` if no `txId` exists in the first place?

Nit: I don't think `this && notation` is very clear, especially when it spreads multiple lines (personal taste though so feel free to ignore).

Reply:
> Need we even call waitForRelayedTx if no txId exists in the first place?

Yes. For when you're relaying the Safe creation.

I agree the pattern is not very elegant. The reason why I has to use it was to reuse the `waitForRelayedTx` to monitor both Safe creation and the modal execution. Any suggestion to improve it here?

Reply:
> The reason why I has to use it was to reuse the waitForRelayedTx to monitor both Safe creation and the modal execution. Any suggestion to improve it here?

As the dispatchers play no role and you are only setting the status on `TaskState.ExecSuccess` or `TaskState.Cancelled`, I would not try to reuse this function. You can then avoid the conditional logic.

Not everything needs to be duplicated, e.g. you can extract `checkTxStatus`.

Reply:
Resolving as we refactored this together.